Orange/widgets/data/owdiscretize.py

```diff
@@ -371,432 +597,421 @@ def update():
-        self.method = Methods.Custom
-        self.cutpoints = points
-        self.manual_cuts_specific.setText(", ".join(map(fmt, points)))
-        self._update_points()
-        self.commit.deferred()
+        """
+        Callback for 'CC' button
 
-    def _copy_to_manual_update_enabled(self):
-        indices = self.selected_indices()
-        methods = [self.var_state[i].method for i in indices]
-        self.copy_current_to_manual_button.setEnabled(
-            len(indices) == 1 and not isinstance(methods[0], Custom))
-
-    def _var_selection_changed(self, *_):
-        self._copy_to_manual_update_enabled()
-        indices = self.selected_indices()
-        # set of all methods for the current selection
-        methods = [self.var_state[i].method for i in indices]
-
-        def key(method):
-            if isinstance(method, Default):
-                return Default, (None, )
-            return type(method), tuple(method)
-
-        mset = list(unique_everseen(methods, key=key))
-
-        self.controlbox.setEnabled(len(mset) > 0)
-        if len(mset) == 1:
-            method = mset.pop()
-            self.method = Methods.from_method(method)
-            if isinstance(method, (EqualFreq, EqualWidth)):
-                self.k = method.k
-            elif isinstance(method, Custom):
-                self.cutpoints = method.points
-        else:
-            # deselect the current button
-            self.method = -1
-            bg = self.controlbox.group
-            button_group_reset(bg)
-        self._update_spin_positions()
-
-    def selected_indices(self):
-        rows = self.varview.selectionModel().selectedRows()
-        return [index.row() for index in rows]
-
-    def method_for_index(self, index):
-        state = self.var_state[index]
-        return state.method
-
-    def discretized_var(self, index):
-        # type: (int) -> Optional[Orange.data.DiscreteVariable]
-        state = self.var_state[index]
-        if state.disc_var is not None and state.points == []:
-            # Removed by MDL Entropy
-            return None
-        else:
-            return state.disc_var
+        Sets selected variables' method to "Custom" and copies thresholds
+        to their VarHints. Variables that are not discretized (for any reason)
+        are skipped.
 
-    def discretized_domain(self):
-        """
-        Return the current effective discretized domain.
+        Discretizations are invalidated and then updated
+        (`_update_discretizations`).
+
+        If all selected variables have the same thresholds, it copies it to
+        the line edit. Otherwise it unchecks all radio buttons to keep the
+        interface consistent.
         """
-        if self.data is None:
-            return None
+        varkeys = self.varkeys_for_selection()
+        texts = set()
+        for key in varkeys:
+            dvar = self.discretized_vars.get(key)
+            fmt = self.data.domain[key[0]].repr_val
```

Comment:
There are at least two ways this can crash:
1) `self.data` is None
`AttributeError: 'NoneType' object has no attribute 'domain'`
Reproduce: Add Discretize widget, open and click CC
2) `key` is DefaultKey/None (see L744)
`TypeError: 'NoneType' object is not subscriptable`
Reproduce: Connect some data to Discretize widget, click CC

Reply:
https://github.com/biolab/orange3/pull/5919/commits/86b10c09c04fb66f04454f98826eb67de5e3117c should now fix it. The button should be disabled when the user selects "Default settings", and Default settings should be selected on "Set data" (which selects it when there is no data).